src/lensFind.js

```diff
@@ -0,0 +1,36 @@
+var __ = require('./__');
+var _curry3 = require('./internal/_curry3');
+var findIndex = require('./findIndex');
+var map = require('./map');
+var update = require('./update');
+
+
+/**
+ * Returns a lens whose focus is the first element that matches the predicate
+ *
+ * @func
+ * @memberOf R
+ * @category
+ * @typedefn Lens s a = Functor f => (a -> f a) -> s -> f s
+ * @sig (a -> Boolean) -> Lens s a
+ * @param {Function} fn The predicate function used to determine if the element is the
+ *        desired one.
+ * @return {Lens}
+ * @see R.view, R.set, R.over
+ * @example
+ *
+ *      var xs = [{a: 1}, {a: 2}, {a: 3}];
+ *      var xLens = R.lensFind(R.propEq('a', 1));
+ *
+ *      R.view(xLens, xs);                    //=> {a: 1}
+ *      R.set(xLens, {b: 4}, xs);             //=> [{b: 4}, {a: 2}, {a: 3}]
+ *      R.over(xLens, R.assoc('a', -1), xs);  //=> [{a: -1}, {a: 2}, {a: 3}]
+ */
+
+module.exports = _curry3(function lensFind(pred, toFunctorFn, list) {
+  var i = findIndex(pred, list);
+  return map(
+    update(i, __, list),
+    toFunctorFn(list[i])
+  );
+});
```

Comment:
One nit, and one more serious concern.

The nit is that we generally use `idx` not `i` for such variables.

The bigger concern has to do with when the value is *not* found.  I don't know enough of the theory behind lenses to understand this note from @scott-christopher: 

> (note: this will only be a valid lens if it is guaranteed to match at least one element of the list)

I'm not really sure what makes for a "valid lens".  But this implementation (and Scott's as well?) would have problems if there isn't a match:

```js
const xLens = lensFind(R.propEq('a', 5));
set(xLens, {b: 20}, [{a: 0}, {a: 1}, {a: 2}, {a: 3}])
//=> [{"a": 0}, {"a": 1}, {"a": 2}, {"b": 20}]
```

... since `update` uses negative indices as counting from the end, but `findIndex` returns `-1` as the result of not finding a value.  Perhaps something like this would be worth doing:

```js
  return map(
    idx > -1 ? update(idx, __, list) : always(list),
    toFunctorFn(list[idx])
  );
```

But I have no idea how that fits with the theory of what makes a valid lens.